### PR TITLE
Reject X-FORWARDED-* headers if processing is disabled

### PR DESCRIPTION
**Pull request by wendigo**
Fixes https://github.com/trinodb/trino/issues/6552

### CODE REVIEW

File: http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        if (request.getHeaders().stream().anyMatch(RejectForwardedRequestCustomizer::isForwardingHeader)) {
+            throw new BadMessageException(HttpStatus.NOT_ACCEPTABLE_406, "Forwarded header is forbidden");
+        }
+        return request;
+    }
+
+    private static boolean isForwardingHeader(HttpField httpField)
+    {
+        return httpField.getLowerCaseName().startsWith(X_FORWARDED_PREFIX);
```
Comment (findepi) [inline http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java:26]: `httpField.getName().regionMatches(true, 0, X_FORWARDED_PREFIX, 0, X_FORWARDED_PREFIX.length())` might be more efficient, by avoiding lowercasing (`getLowerCaseName` returns precomputed value for some headers, but not all)

File: http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+public class RejectForwardedRequestCustomizer
+        implements HttpConfiguration.Customizer
+{
+    private static final String X_FORWARDED_PREFIX = "x-forwarded-";
+
+    @Override
+    public Request customize(Request request, HttpFields.Mutable responseHeaders)
+    {
+        if (request.getHeaders().stream().anyMatch(RejectForwardedRequestCustomizer::isForwardingHeader)) {
+            throw new BadMessageException(HttpStatus.NOT_ACCEPTABLE_406, "Forwarded header is forbidden");
```
[the remark this answers is not in the document]
Comment (findepi) [reply http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java:19]: would be good to provide exact header name in the message?

**Review by findepi — APPROVED** [review]

File: http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java (hunk 1 above)
Comment (wendigo) [reply http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java:26]: Nice!

File: http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (wendigo) [reply http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java]: Done. PTAL

**Review by findepi — APPROVED** [review]

File: http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Override
+    public Request customize(Request request, HttpFields.Mutable responseHeaders)
+    {
+        Optional<HttpField> forbiddenHeader = request.getHeaders().stream()
+                .filter(RejectForwardedRequestCustomizer::isForwardingHeader)
+                .findFirst();
+
+        if (forbiddenHeader.isPresent()) {
+            throw new BadMessageException(HttpStatus.NOT_ACCEPTABLE_406, "Server configuration does not allow processing of %s header".formatted(forbiddenHeader.get().getName()));
```
[the remark this answers is not in the document]
Comment (wendigo) [reply http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java]: Done.

**Pull request merged**